hack/jsonschema/openapi2jsonschema.go

```diff
@@ -0,0 +1,109 @@
+//nolint:forbidigo
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+func extractSchema(targetDir string, spec map[string]interface{}) {
+	if versions, ok := spec["versions"].([]interface{}); ok {
+		for _, version := range versions {
+			versionData := version.(map[string]interface{})
+			if schema, ok := versionData["schema"].(map[string]interface{}); ok {
+				if openAPIV3Schema, ok := schema["openAPIV3Schema"].(map[string]interface{}); ok {
+					processAndWriteSchema(targetDir, openAPIV3Schema, spec, versionData["name"].(string))
+				}
+			}
+		}
+	}
+}
+
+func processAndWriteSchema(targetDir string, schema map[string]interface{}, spec map[string]interface{}, versionName string) {
+	schema = additionalProperties(schema, true)
+	schema = replaceIntOrString(schema)
+
+	name := spec["names"].(map[string]interface{})["plural"].(string)
+	group := spec["group"]
+
+	filename := fmt.Sprintf("%s_%s_%s.json", group, name, versionName)
+	filename = strings.ToLower(filename)
+	writeSchemaFile(schema, filepath.Join(targetDir, filename))
+}
+
+func additionalProperties(data map[string]interface{}, skip bool) map[string]interface{} {
+	if _, exists := data["properties"]; exists && !skip {
+		if _, exists := data["additionalProperties"]; !exists {
+			data["additionalProperties"] = false
+		}
+	}
+
+	for _, v := range data {
+		if subData, ok := v.(map[string]interface{}); ok {
+			additionalProperties(subData, false)
+		}
+	}
+
+	return data
+}
+
+func replaceIntOrString(data map[string]interface{}) map[string]interface{} {
+	if format, exists := data["format"]; exists && format == "int-or-string" {
+		data["oneOf"] = []interface{}{
```

Comment:
I don't see any `oneOf` in the generated files but there are some `anyOf`. I think there is a slight difference between the two. `oneOf` is a `XOR` while `anyOf` is a `OR` which IMO makes more sense